doc/CMakeLists.txt

```diff
@@ -188,7 +188,7 @@ function(add_doc TARGET SOURCE DEPS)
             ${CMAKE_CURRENT_SOURCE_DIR}/script/page-list.awk
             ${TARGET}.ps.tmp
             ${TARGET}.ps
-        COMMAND ${PS2PDF} ${TARGET}.ps ${TARGET}
+        COMMAND sh ${PS2PDF} ${TARGET}.ps ${TARGET}
```

Comment:
This is the way it worked before latest refactorings, but as mentioned in the commit message, it only works if ps2pdf is a shell script.

I tried the following in `etc/configure.cmake`:
```
if(WIN32)
  execute_process(
    COMMAND ${CYGPATH} -u "${PS2PDF}"
    OUTPUT_VARIABLE PS2PDF
    ERROR_QUIET)
endif()
```
But it resulted in `/mingw64/bin/ps2pdf` instead of `/c/xxx/msys64/mingw64/bin/ps2pdf` (which was the output when used directly on the "MSYS2 MinGW x64" command line.

Reply:
Sorry for breaking the build on MSYS2.
I assumed that explicit `sh` is needed for self-written scripts on MSYS2. So kept them, but accidentally lost the one for ref-ptx on rebase.

But I removed the `ps2pdf` on purpose (and forgot to document it), because it's an installed program. For such we shall not assume if it's sh, perl, python... 

`find_program` (in contrast to `find_file`) searches only for executables. So if that one is not executable, that's the problem of that find. Do you call a MSYS2-native CMake and ninja/make, or a normal Windows version? Do you call it inside a MSYS2 shell or directly from outside?

Reply:
I do not understand MSYS2, I thought it's main purpose is to run typical unix builds on Windows without changing makefiles.

I think I found an acceptable solution which avoids knowledge about programs.
Remove the `find_program` completely for now and use `COMMAND sh -c "ps2pdf ${TARGET}.ps ${TARGET}"`.
As `groff` and the other programs are currently also not checked for their existence, this is no step backwards.

I just can't test it on MSYS2, because in the workflow soelim fails as it cannot find the generate .so files.

Reply:
> But I removed the `ps2pdf` on purpose (and forgot to document it), because it's an installed program. For such we shall not assume if it's sh, perl, python...

I guessed so and it totally makes sense.

> `find_program` (in contrast to `find_file`) searches only for executables. So if that one is not executable, that's the problem of that find.

No, `ps2pdf` is executable and `find_program` works correctly resulting in `C:\xxx\msys64\mingw64\bin\ps2pdf`. But I see it now (after some time away from keyboard):
All commands are executed via `cmd.exe` and despite having listed the broken/fixed commands in the commit message I didn't fully realize that apparently. Of course `cmd` cannot execute a shell script directly and needs to call `sh` with the script file as parameter. Converting the path via `cygpath` as I tried was just useless here.

Maybe it can be considered to be a bug or at least a shortcoming in cmake? Or this is a common problem (I'm sure) with a proper solution already.
But as long as I don't know about another solution (and it's hard to find the proper words for searching for it) there is no way around that `sh`. Will change that commit to only go this way for WIN32.

> Do you call a MSYS2-native CMake and ninja/make, or a normal Windows version? Do you call it inside a MSYS2 shell or directly from outside?

My setup is as described in BUILDING.pdf. I'm inside the "MSYS2 MinGW x64" command window - that's running Bash and preferring the native mingw64 commands (come first in PATH) over the msys2 ones. All tools are installed from the MSYS2 package repository via pacman. `cmake.exe`, `ninja.exe` and `mingw32-make.exe` are all mingw64 builds.

Reply:
Ok, so it's the same as in the github worklow.
Can you try out the `COMMAND sh -c "ps2pdf ${TARGET}.ps ${TARGET}"` as suggested above.

I need to setup an MSYS2 in a virtual machine myself to try it out. Then I could ask the right questions on CMake's discourse. Based on this [comment](https://discourse.cmake.org/t/how-to-check-that-the-current-compiler-is-gcc-msys-when-using-ninja-generator/6698/8), I thought CMake takes care of such things.

Reply:
OK, so I have adapted this commit now.
Similar to the formerly existing PS2PDFEXE, but reusing the PS2PDF variable.

> Can you try out the `COMMAND sh -c "ps2pdf ${TARGET}.ps ${TARGET}"` as suggested above.

It worked as well, both with `Ninja` and `MinGW Makefiles` generators.